p2p/kademlia/dao/sqlite/queries.go

```diff
@@ -0,0 +1,172 @@
+package sqlite
+
+import (
+	"context"
+	"database/sql"
+	"time"
+
+	// add sqlite driver
+	_ "github.com/mattn/go-sqlite3"
+	"github.com/pastelnetwork/gonode/common/log"
+	"github.com/pastelnetwork/gonode/p2p/kademlia/crypto"
+)
+
+var (
+	logPrefix = "p2p"
+)
+
+// Migrate runs migrations such as creating `keys` table.
+func Migrate(ctx context.Context, db *sql.DB) error {
+	ctx = log.ContextWithPrefix(ctx, logPrefix)
+	ctx, cancel := context.WithDeadline(ctx, time.Now().Add(3*time.Second))
+	defer cancel()
+
+	_, err := db.ExecContext(ctx,
+		"CREATE TABLE IF NOT EXISTS `keys` (`uid` INTEGER PRIMARY KEY AUTOINCREMENT, `key` VARCHAR(64) NULL, `data` BLOB NULL, `replication` DATE NULL, `expiration` DATE NULL)")
+	if err != nil {
+		log.WithContext(ctx).Infof("Error %s when creating keys table", err)
+		return err
+	}
+
+	return nil
+}
+
+// Store will store a key/value pair for the local node with the given
+// replication and expiration times.
+func Store(ctx context.Context, db *sql.DB, data []byte, replication, expiration time.Time) (int64, error) {
+	key := crypto.GetKey(data)
+	query := "INSERT INTO keys(key, data, replication, expiration) VALUES (?, ?, ?, ?)"
+	ctx = log.ContextWithPrefix(ctx, logPrefix)
+	ctx, cancel := context.WithDeadline(ctx, time.Now().Add(3*time.Second))
+	defer cancel()
+
+	stmt, err := db.PrepareContext(ctx, query)
+	if err != nil {
+		log.WithContext(ctx).Infof("Error %s when preparing SQL statement", err)
+		return 0, err
+	}
+	defer stmt.Close()
+
+	res, err := stmt.ExecContext(ctx, string(key), data, replication, expiration)
+	if err != nil {
+		log.WithContext(ctx).Infof("Error %s when inserting row into keys table", err)
+		return 0, err
+	}
+
+	rows, err := res.RowsAffected()
+	if err != nil {
+		log.WithContext(ctx).Infof("Error %s when finding rows affected", err)
```

Comment:
`Info` level ?
and not only in this place

Reply:
oh, ok i need to use `log.WithContext(ctx).Fatalf`

Reply:
updated.